build.sbt

```diff
@@ -400,9 +413,16 @@ lazy val sbtScalaNative =
             testRunner / publishLocal
           )
           .value
-      }
+      },
+      // Depending on projects defining `crossScalaVersion` other then this `scalaVersion` would result in build failure
+      unmanagedSourceDirectories in Compile ++= Seq(
+        (testRunner / Compile / scalaSource).value,
+        (tools / Compile / scalaSource).value,
+        (nir / Compile / scalaSource).value,
+        (util / Compile / scalaSource).value,
+        baseDirectory.value.getParentFile / "test-interface-common/src/main/scala"
```

Comment:
We must not do this. That would mean that the published artifacts have two copies of those things.

We should keep the regular `dependsOn` dependency on the other projects. Instead, we should avoid running any task in `sbtScalaNative/*` when we have done `++2.11.*` or `++2.13.*`.

Reply:
Yes, but then we would not be able to run `sbtScalaNative/scripted` for 2.11 and 2.13.
It would end up with something like this when trying to compile 
```
[error] Modules were resolved with conflicting cross-version suffixes in ProjectRef(uri("file:scala-native/"), "sbtScalaNative"):
[error]    org.scala-native:test-runner _2.12, _2.11
[error]    org.scala-native:nir _2.12, _2.11
[error]    org.scala-native:tools _2.12, _2.11
```

It might be a bug or design flow in sbt. 
How about duplicating these 3 projects, eg. making project `pluginTools` using the same directory as `tools` but with other configurations? May it actually work? In Scala.js thanks to duplication of projects there is no such issue because you explicitly command to depend on `*v2_12` project.

Reply:
> Yes, but then we would not be able to run `sbtScalaNative/scripted` for 2.11 and 2.13.

Right. I think we will have to live with that. Anyway, it doesn't really make sense; not in the way `++` is supposed to be used. If I do `; ++ 2.13.4 ; sbtScalaNative/scripted`, I would *expect* that sbt-scala-native itself be compiled with 2.13.4, and then we run `scripted` on that. Of course that doesn't work because an sbt plugin cannot build on 2.13.4. It seems that so far we've abused `++` and `scripted` to mean "compile the sbt plugin with 2.12 anyway, but inside the scripted tests, use projects with version 2.13.4".

> How about duplicating these 3 projects, eg. making project `pluginTools` using the same directory as `tools` but with other configurations? May it actually work? In Scala.js thanks to duplication of projects there is no such issue because you explicitly command to depend on `*v2_12` project.

Eventually, we'll probably have to decompose into `v2_*` like in Scala.js. But that has a significant cost on build load time. The ability to run scripted tests for 2.11 and 2.13 is not worth that additional time. In Scala.js we had to do it to be able to remove the `linker` from the *static* dependencies of `sbt-scalajs`, and instead dynamically class-load it. But we're not there yet for Scala Native.

Reply:
Actually, I've found a way around that would satisfy both using `dependsOn` and remaining scriptedTests for 2.11 and 2.13. 
I've extracted scripted tests to separate project with enabled `ScriptedPlugin` and locked scalaVersion to 2.12. 
This allows publishing artifacts based on `scalaVersion` defined in `ThisBuild`. The remaining problem was publishing plugin artifact itself - it is done in alias `test-scripted` - in first command we run `+ sbtScalaNative/publishLocal` - this way it would publish for all (so 2.12 only) plugin versions. 

I needed to move scripted sources to the inner directory to make sure that they don't interfere with `target` which would lead to failure, so I may add this change in follow-up PR.